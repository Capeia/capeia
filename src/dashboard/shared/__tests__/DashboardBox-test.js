// @flow
/* eslint-env jest, jasmine */

import React from 'react'
import renderer from 'react-test-renderer'

describe('<DashboardBox>', () => {
  it('renders correctly', () => {
    const Component = require('../DashboardBox').default
    const instance = renderer.create(<Component>Hello World</Component>)
    let tree = instance.toJSON()
    expect(tree).toMatchSnapshot()

    instance.update(<Component title='Hello'>World</Component>)
    tree = instance.toJSON()
    expect(tree).toMatchSnapshot()

    instance.update(<Component title='Hello' description='World' />)
    tree = instance.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
