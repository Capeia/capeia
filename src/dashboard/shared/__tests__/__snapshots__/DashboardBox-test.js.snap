// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardBox> renders correctly 1`] = `
<div
  className="DashboardBox"
>
  <div
    className="content"
  >
    Hello World
  </div>
</div>
`;

exports[`<DashboardBox> renders correctly 2`] = `
<div
  className="DashboardBox"
>
  <header>
    <h1>
      Hello
    </h1>
    <p
      className="description"
    />
  </header>
  <div
    className="content"
  >
    World
  </div>
</div>
`;

exports[`<DashboardBox> renders correctly 3`] = `
<div
  className="DashboardBox"
>
  <header>
    <h1>
      Hello
    </h1>
    <p
      className="description"
    >
      World
    </p>
  </header>
  <div
    className="content"
  />
</div>
`;
